### DISCUSSION TITLE
brew postinstall gcc loop

### DISCUSSION THREAD

**Discussion by octocatsup** [Homebrew/discussions#2954] [Linux] [answered]
I am trying to install brew on Ubuntu > WSL2, which works, but I cannot get some packages to work. AFAIK, it has something to do with gcc or isl because those are the packages giving me issues. I ran "brew update && brew upgrade" to update packages, ran "brew doctor" to fix issues. Requests that I find a replacement for isl@0.18 and reinstall gcc. I have tried both separately, together, and every way I can think to do so. Nothing works. brew shows they're installed when I use "brew list," and here is my config, which also shows gcc@5:

HOMEBREW_VERSION: 3.3.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: d603fd6426ddae27327bb51afb085653eb2d84a7
Last commit: 11 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 206f489a72728bf9dbb207ffd1d04d7654c1c63a
Core tap last commit: 13 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: quad-core 64-bit skylake
Clang: N/A
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /bin/curl
Kernel: Linux 4.4.0-19041-Microsoft x86_64 GNU/Linux
OS: Ubuntu 20.04.3 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: N/A
/usr/bin/ruby: N/A
glibc: N/A
gcc@5: 5.5.0_7
xorg: N/A

Running "brew doctor," brew requests that I reinstall gcc, which I do, and then it fails to complete postinstall tasks. It requests that I run "brew postinstall gcc" and fails, requests that I run "brew postinstall gcc" and on the loop goes. Not sure how to proceed... 🤷‍♂️

**Comment by carlocab** [comment, marked as answer]
This is your problem:
```
/usr/bin/gcc: N/A
```
Do
```
sudo apt-get install build-essential procps file
brew reinstall gcc gcc@5
```
See https://docs.brew.sh/Homebrew-on-Linux for details.

**Reply by octocatsup** [reply]
@carlocab Nailed it! Thank you very much for the insanely fast help 😊

I have one more quick question. Do I need gcc@5 and isl@0.18, since I have gcc/isl current packages? I am only asking because I receive a warning about isl@0.18 being out of date.

**Reply by octocatsup** [reply]
Actually, nvm. It looks like gcc@5 is the current version. I thought it was 11 for some reason.

**Reply by carlocab** [reply]
You can probably uninstall `gcc@5` and `isl@0.18`. You only need it if your host compiler is older than GCC 5, but your `/usr/bin/gcc` should be newer than that.
